### ISSUE TITLE
[EPIC] Deltas between designs and build for join token gated community flow

### ISSUE THREAD

**Issue by benjthayer** [status-im/status-desktop#13988]
Below is a list of differences between the intended user flow for joining a token gated community and what is currently in the app (2.27.0-rc.6)

**Screen recording of app:**
https://github.com/status-im/status-desktop/assets/110033914/e6805cf8-71c1-4932-ab64-fdca402fec52

**Differences:**

-----

**1) Join CTA labelling**

App: `Reveal your address and request to join` (with authentication icon)

<img width="1584" alt="Screenshot 2024-03-15 at 9 32 02 AM" src="https://github.com/status-im/status-desktop/assets/110033914/4f26464d-ba05-4b29-b50f-5df62e281e58">

Design: `Request to join community` (no authentication icon - there is no authentication at this point)

<img width="1232" alt="Screenshot 2024-03-15 at 9 33 39 AM" src="https://github.com/status-im/status-desktop/assets/110033914/1142b0c1-f03b-48b8-96f5-e4a8c6d4963c">

-------

Additional note:
The n members should be pluralised if >1 but not if only 1. I.e. we should never see `member(s)` here only `member` or `members` depending on the number of members.

<img width="305" alt="Screenshot 2024-03-15 at 9 34 55 AM" src="https://github.com/status-im/status-desktop/assets/110033914/937d3ba1-961d-4138-9b6c-6b2935898a70">

-------

**2) Request to join dialog** 

App: Dialog title reads `Welcome to [community name]`

<img width="679" alt="Screenshot 2024-03-15 at 9 36 35 AM" src="https://github.com/status-im/status-desktop/assets/110033914/cfa031ae-0c26-445d-a3c3-cfbed13d8dfe">

Design: Dialog title reads `Request to join [community name]`

<img width="574" alt="Screenshot 2024-03-15 at 9 38 01 AM" src="https://github.com/status-im/status-desktop/assets/110033914/fd60f7d5-9ec3-4887-a11f-93cf437a70e4">

App: Dialog primary CTA reads `Join [community name]` (without authentication icon)
Design: Dialog primary CTA reads `Share all addresses to join` (with authentication icon)

App: Missing `eligible to join` tag
Design: Contains an `eligible to join` tag sticky to the base of the dialog with the relevant highest possible role the user can join the community as based on their aggregated token balances across all of their Status accounts

-------

Additional note:
Community icon looks a little blurry

-------

**3) Select addresses dialog**

App: 
- Unknown rationale for account ordering?
- The account which holds the relevant collectible for joining the community doesn't show the collectible under the account name
- When the user checks accounts on and off, the UI in the backgrounds seems to shift / change according to the selections.
- The `Requirements check pending` text resizes the dialog when it is shown each time an account is selected/deselected which is a little distracting. Perhaps one for design to take a look at. 
- The `Share selected addresses to join` should have an authentication icon in the case where the user is opting to share addresses that are all stored in the same location because the user can authenticate a single time for all accounts. The CTA should omit the authentication icon if the relevant key pairs are stored across multiple locations (i.e. a mix of keycards or a mix of device/keycards). In this case, the authentication wiki flow should launch. The flows showing this logic are here:

https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44966-272706&mode=design&t=EBMjLYoaJAQORTJz-4

<img width="662" alt="Screenshot 2024-03-15 at 9 52 42 AM" src="https://github.com/status-im/status-desktop/assets/110033914/f7a77e95-5ce5-4cd2-ab33-15772c77712d">

Design: 
- Account ordering should be according to balance of relevant tokens for joining the community/channels. If there are no relevant tokens across any of the accounts, the accounts should be ordered according to the user's wallet ordering. 

<img width="542" alt="Screenshot 2024-03-15 at 9 42 56 AM" src="https://github.com/status-im/status-desktop/assets/110033914/5676fe15-34ba-4c2f-8fe2-113a389f25e3">

-------

**4) Prove ownership of key pairs dialog**

App: 
The user should not see this dialog if all key pairs for the selected accounts are stored in the same location. For example in this flow, all of the accounts I wish to share with the community are stored on device and therefore the primary CTA on the `Selected addresses to share with [community name]` dialog should feature an authentication icon - clicking that should launch the authentication dialog not the `Prove ownership of key pairs` dialog. The `Prove ownership of key pairs` dialog should only launch if the key pairs associated with the accounts the user wishes to share with the community are stored in different locations and therefore require individual authentication.

Also when the user clicks the `Authenticate` button on the `Authenticate` dialog, there is a long pause 0:49-0:55 in the video. If the authentication process takes time, we should add a spinner state to the `Authenticate` button so the user knows something is happening and not to move away.

<img width="663" alt="Screenshot 2024-03-15 at 10 03 44 AM" src="https://github.com/status-im/status-desktop/assets/110033914/afda081e-2418-45b0-ad8a-bb3726b23f5d">

Design:

<img width="630" alt="Screenshot 2024-03-15 at 10 04 11 AM" src="https://github.com/status-im/status-desktop/assets/110033914/879eec7a-97b5-43a1-8544-e218c3baf159">

-------

**5) After authentication**

App: Once the user has authenticated the relevant keypair/s, the user is returned to this screen state where the `Reveal your address and request to join` button is still shown:

<img width="1586" alt="Screenshot 2024-03-15 at 10 07 25 AM" src="https://github.com/status-im/status-desktop/assets/110033914/b50c6206-5776-44db-9039-4a8a3e689c87">

Design: Once the user has authenticated, the button should change to `Membership Request Pending...`. Clicking on this CTA should relaunch the `Request to join` dialog - the only CTA on the `Request to join dialog` should be `Cancel Membership Request`

<img width="1072" alt="Screenshot 2024-03-15 at 10 10 13 AM" src="https://github.com/status-im/status-desktop/assets/110033914/eb6d2d2d-da22-41cb-94ba-ce94cf7fc25e">

<img width="1066" alt="Screenshot 2024-03-15 at 10 09 59 AM" src="https://github.com/status-im/status-desktop/assets/110033914/e7d7ccd7-b0f5-4c10-8397-29d66fa4441e">

https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2365-302551&mode=design&t=EBMjLYoaJAQORTJz-4

cc @John-44 @iurimatias @jrainville 

-------

## Subtasks UI-TEAM:

**JOIN / GLOBAL / MAIN VIEW:**
- [x] https://github.com/status-im/status-desktop/issues/14098
- [x] https://github.com/status-im/status-desktop/issues/14099
- [x] https://github.com/status-im/status-desktop/issues/14097
- [x] https://github.com/status-im/status-desktop/issues/14104
- [x] https://github.com/status-im/status-desktop/issues/14077
- [x] https://github.com/status-im/status-desktop/issues/14179

**REQUEST TO JOIN DIALOG**
- [x] https://github.com/status-im/status-desktop/issues/14100

**SHARED ADDRESSES POPUP / PANEL:**
- [x] https://github.com/status-im/status-desktop/issues/14101
- [x] https://github.com/status-im/status-desktop/issues/14102
- [x] https://github.com/status-im/status-desktop/issues/14103
- [x] https://github.com/status-im/status-desktop/issues/14096

**Comment by jrainville** [comment]
@alaibe @alexandraB99 it might be worth splitting this task in 2. Sale can keep working on https://github.com/status-im/status-desktop/issues/13974 and the UI team can do the other UI element fixes listed above

**Comment by caybro** [comment]
Agreed with Sale he'd do points 3. and 4., the rest is on me

**Comment by noeliaSD** [comment]
We are going to consider this task as an EPIC and I'm going to create smaller tasks to work in parallel:
- [ ] https://github.com/status-im/status-desktop/issues/14090

**Comment by saledjenic** [comment]
A short update, at least points 3 and 4 are fully handled in https://github.com/status-im/status-desktop/pull/14024

**Comment by noeliaSD** [comment]
Closing the epic since all the important tasks have been closed and the missing ones are postponed to other milestones bc they have lower priority.
